packages/ui/react-ui-table/src/model/table-controls.ts

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type MouseEvent } from 'react';
+
+import { type TableModel } from './';
+import { ModalController } from './modal-controller';
+import { tableControls } from '../util/table-controls';
+
+export class TableControls {
```

Comment:
i don't understand this and i think it's just exporting what was currently a very leaky abstraction from model into another model (which might be incremental progress, but still is covering up a crime).

IMO event handlers should not be moved out of the class that owns the UX. E.g., the following is good:

```ts
cons MyComponent = ({ item, onSelect }) => {
  const handleClick = (ev) => onSelect(item);
  return <div onClick={handleClick} />
}
```

i.e., "click" (and any subsequent event type) is an implementation detail of the component implementation. Its callbacks should abstract that and involve higher level semantics -- and almost never expose events, unless they are also in the intimate business of handling the specific UX implementation.

here, this is just plumbing that seems to add another level of indirection. how would you test this class?

i think these handlers would be better off inside the table itself.

Reply:
Resolved! I agree and you make some good points.

I've done a cleanup pass over all the click handling / modal buttons. Moved the event handler into the Table component.